### PR TITLE
Compose Map InteractionOptions

### PR DESCRIPTION
**Pull request by shubham7109**
Related issue: `#2964`

### Description
PR to accept `MapViewInteractionOptions` as a parameter to the compose Map

### Testing
```kotlin
val arcGISMap by remember { mutableStateOf(ArcGISMap(BasemapStyle.ArcGISStreets)) }
val interactionOptions by remember { mutableStateOf(MapViewInteractionOptions()) }

Map(
    modifier = Modifier.fillMaxSize(),
    arcGISMap = arcGISMap,
    mapViewInteractionOptions = interactionOptions
)

Button(onClick = {
    interactionOptions.isPanEnabled = false
}) {
    Text(text = "Disable pan")
}
```

### CODE REVIEW

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt [hunk 1]
```diff
@@ -43,14 +44,19 @@ import com.arcgismaps.mapping.view.MapView
 public fun Map(
     modifier: Modifier = Modifier,
     arcGISMap: ArcGISMap? = null,
+    mapViewInteractionOptions: MapViewInteractionOptions? = null,
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt:47]: Semantically it's not quite correct that interaction options are nullable, which implies "absence of interaction options". But the MapView will always operate on some interaction options, i.e. default interaction options to start with. To follow [Compose API guidelines](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-component-api-guidelines.md#componentdefault-object), I think we should define a `MapViewInteractionOptionDefaults` const and use that to initialize the parameter:

```kotlin
public val MapViewInteractionOptionDefaults = MapViewInteractionOptions()

...

@Composable
fun Map(
   arcGISMap: ArcGISMap? = null,
   ...
   mapViewInteractionOptions: MapViewInteractionOptions = MapViewInteractionOptionDefaults,
   ...
) {
}
```

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt [hunk 2]
```diff
@@ -43,14 +44,19 @@ import com.arcgismaps.mapping.view.MapView
...
     overlay: @Composable () -> Unit = {}
 ) {
     val lifecycleOwner = LocalLifecycleOwner.current
     val context = LocalContext.current
     val mapView = remember { MapView(context) }.apply {
         map = arcGISMap
+        if (mapViewInteractionOptions != null) {
+            interactionOptions = mapViewInteractionOptions
+        }
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt:56]: The interaction options should be set in the `update` lambda of the AndroidView. I made that change in the [pending PR for LocationDisplay](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/pull/160/files#diff-ecf595d2aea3485bba4e96790429167cf64dad86aa98708a329029e062fb6fc2R64-R67), which adds the `update` lambda. You could picky-bag on that PR.

**Review by gunt0001 — COMMENTED** [review]
@shubham7109 - see my comments.

**Comment by shubham7109** [comment]
@gunt0001 Added PR feedback changes, ready for review.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt [hunk 3]
```diff
@@ -30,6 +30,9 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.viewinterop.AndroidView
 import com.arcgismaps.mapping.ArcGISMap
 import com.arcgismaps.mapping.view.MapView
+import com.arcgismaps.mapping.view.MapViewInteractionOptions
+
+public val MapViewInteractionOptionDefaults: MapViewInteractionOptions = MapViewInteractionOptions()
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt:35]: missing doc

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt [hunk 4]
```diff
@@ -43,21 +46,25 @@ import com.arcgismaps.mapping.view.MapView
 public fun Map(
     modifier: Modifier = Modifier,
     arcGISMap: ArcGISMap? = null,
+    mapViewInteractionOptions: MapViewInteractionOptions = MapViewInteractionOptionDefaults,
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt:49]: Needs an `@param` in the doc above

**Review by gunt0001 — APPROVED** [review]
@shubham7109 - apart from missing doc this looks good.

**Comment by shubham7109** [comment]
Thanks @gunt0001, I've added the missing doc and updated this branch with the base branch. Ready for review @puneet-pdx

**Review by puneet-pdx — APPROVED** [review]
👍🏽

**Pull request merged**